extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/YoutubeParsingHelper.java

```diff
@@ -347,23 +396,27 @@ private static void extractClientVersionAndKey() throws IOException, ExtractionE
 
         try {
             key = Parser.matchGroup1("INNERTUBE_API_KEY\":\"([0-9a-zA-Z_-]+?)\"", html);
-        } catch (Parser.RegexException e) {
+        } catch (final Parser.RegexException e1) {
             try {
                 key = Parser.matchGroup1("innertubeApiKey\":\"([0-9a-zA-Z_-]+?)\"", html);
-            } catch (Parser.RegexException ignored) {
+            } catch (final Parser.RegexException e2) {
+                keyAndVersionExtracted = false;
+                throw new ParsingException("Could not extract client version and key");
             }
         }
+        keyAndVersionExtracted = true;
     }
 
     /**
      * Get the client version
      */
     public static String getClientVersion() throws IOException, ExtractionException {
         if (!isNullOrEmpty(clientVersion)) return clientVersion;
-        if (isHardcodedClientVersionValid()) return clientVersion = HARDCODED_CLIENT_VERSION;
+        if (areHardcodedClientVersionAndKeyValid()) {
+            return clientVersion = HARDCODED_CLIENT_VERSION;
+        }
 
-        extractClientVersionAndKey();
-        if (isNullOrEmpty(key)) throw new ParsingException("Could not extract client version");
+        if (!keyAndVersionExtracted) extractClientVersionAndKey();
```

Comment:
The `if (!keyAndVersionExtracted) ` check here is unneded, as it is already present in the second line of `extractClientVersionAndKey()`